homeassistant/components/mqtt/remote.py

```diff
@@ -0,0 +1,183 @@
+    hass: HomeAssistantType, config: ConfigType, async_add_entities, discovery_info=None
+):
+    """Set up MQTT remote through configuration.yaml."""
+    await _async_setup_entity(config, async_add_entities, discovery_info)
+
+
+async def _async_setup_entity(config, async_add_entities, config_entry=None):
+    """Set up the MQTT remote."""
+    async_add_entities([MqttRemote(config, config_entry)])
+
+
+# pylint: disable=too-many-ancestors
+class MqttRemote(
+    MqttAttributes, MqttAvailability, MqttEntityDeviceInfo, RemoteDevice, RestoreEntity
+):
+    """Representation of a remote that can be controlled using MQTT."""
+
+    def __init__(self, config, config_entry):
+        """Initialize the MQTT remote."""
+        self._state = False
+        self._sub_state = None
+
+        self._state_on = None
+        self._state_off = None
+        self._optimistic = None
+        self._unique_id = config.get(CONF_UNIQUE_ID)
+
+        # Load config
+        self._setup_from_config(config)
+
+        device_config = config.get(CONF_DEVICE)
+
+        MqttAttributes.__init__(self, config)
+        MqttAvailability.__init__(self, config)
+        MqttEntityDeviceInfo.__init__(self, device_config, config_entry)
+
+    async def async_added_to_hass(self):
+        """Subscribe to MQTT events."""
+        await super().async_added_to_hass()
+
+    def _setup_from_config(self, config):
+        """(Re)Setup the entity."""
+        self._config = config
+        self._optimistic = config[CONF_OPTIMISTIC]
+        self._commands = config.get(CONF_COMMANDS)
+
+    async def async_will_remove_from_hass(self):
+        """Unsubscribe when removed."""
+        self._sub_state = await subscription.async_unsubscribe_topics(
+            self.hass, self._sub_state
+        )
+        await MqttAttributes.async_will_remove_from_hass(self)
+        await MqttAvailability.async_will_remove_from_hass(self)
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the remote."""
+        return self._config[CONF_NAME]
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    @property
+    def assumed_state(self):
+        """Return true if we do optimistic updates."""
+        return self._optimistic
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._unique_id
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._config.get(CONF_ICON)
+
+    async def async_send_command(self, command, **kwargs):
+        """Send a command to the remote.
+
+        This method is a coroutine.
+        """
+        for cmd in command:
+            if cmd in self._commands:
+                self._publish_command(self._commands[cmd][CONF_COMMAND])
+            else:
+                _LOGGER.warning("Unknown command: %s", cmd)
```

Comment:
Why is command a list?
If it should be a list: I would name it `commands`.

Reply:
Since this function is call from outside this class, I don't think I can rename the parameter. If I rename the parameter to commands, it throws an error "TypeError: async_send_command() missing 1 required positional argument: 'commands'".

Reply:
When calling the remote.send_command service in the developer tools, the documentation says, 
"command | A single command or a list of commands to send." So I think this change is outside the scope of this PR.